stdlib/public/core/PrintForDebugger.swift

```diff
@@ -0,0 +1,294 @@
+//===--- PrintForDebugger.swift -------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+public enum _PrintForDebugger {
+
+  enum CollectionStatus {
+    case NotACollection
+    case CollectionOfElements
+    case CollectionOfPairs
+    case Element
+    case Pair
+    case ElementOfPair
+  
+    var isCollection: Bool {
+      return self != .NotACollection
+    }
+  
+    func getChildStatus(child: Mirror) -> CollectionStatus {
+      let disposition = child.displayStyle ?? .struct
+    
+      if disposition == .collection { return .CollectionOfElements }
+      if disposition == .dictionary { return .CollectionOfPairs }
+      if disposition == .set { return .CollectionOfElements }
+    
+      if self == .CollectionOfElements { return .Element }
+      if self == .CollectionOfPairs { return .Pair }
+      if self == .Pair { return .ElementOfPair }
+    
+      return .NotACollection
+    }
+  }
+
+  static func asObjectIdentifier(value: Any) -> ObjectIdentifier? {
+      if let ao = value as? AnyObject {
+          return ObjectIdentifier(ao)
+      } else {
+          return nil
+      }
+  }
+
+  static func asNumericValue(value: Any) -> Int {
+      if let ao = value as? AnyObject {
+          return unsafeBitCast(ao, to: Int.self)
+      } else {
+          return 0
+      }
+  }
+
+  static func asStringRepresentation(value: Any?,
+                              mirror: Mirror,
+                              count: Int) -> String? {
+    let ds = mirror.displayStyle ?? .`struct`
+    switch ds {
+    case .optional:
+        if count > 0 {
```

Comment:
Sorry, unfortunately we don't have anything written down.  Most of the library sources follow the conventions, so when preparing the patch please imitate the other code.  I'd be happy to answer any questions if you are wondering about anything specific.